### PR TITLE
MWPW-129984 Contextual Search without Helpx

### PR DESCRIPTION
**Pull request by Brandon32**
* Prevent Enter from creating a new tab when advanced search is not set.
* Add no results placeholder.

Resolves: [MWPW-129984](https://jira.corp.adobe.com/browse/MWPW-129984)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/bmarshal/search/contextual-search?martech=off
- After: https://bmarshal-search-results--milo--adobecom.hlx.page/drafts/bmarshal/search/contextual-search?martech=off

**Bacom Blog**
- Before: https://main--bacom-blog--adobecom.hlx.page/drafts/bmarshal/?martech=off
- After: https://main--bacom-blog--adobecom.hlx.page/drafts/bmarshal/?martech=off&milolibs=bmarshal-search-results

**Blog**
- Before: https://main--blog--adobecom.hlx.page/?martech=off
- After: https://main--blog--adobecom.hlx.page/?martech=off&milolibs=bmarshal-search-results

### CODE REVIEW

File: libs/blocks/gnav/gnav-contextual-search.js [hunk 1]
```diff
@@ -97,11 +97,16 @@ export default async function onSearchInput({ value, resultsEl, searchInputEl, a
 
   if (currentSearch === lastSearch) {
     if (!hits.length) {
-      const advancedLink = advancedSearchEl.querySelector('a');
-      const href = new URL(advancedLink.href);
-      href.searchParams.set('q', value);
-      advancedLink.href = href.toString();
-      resultsEl.replaceChildren(advancedSearchEl);
+      const emptyMessage = createTag('p', {}, 'No results');
```
Comment (meganthecoder) [inline libs/blocks/gnav/gnav-contextual-search.js:100]: Should "no results" be a placeholder value so it can be translated?

Comment (Brandon32) [reply libs/blocks/gnav/gnav-contextual-search.js]: Updated, thanks!

**Review by meganthecoder — APPROVED** [review]

**Review by mokimo — APPROVED** [review]

File: libs/blocks/gnav/gnav-contextual-search.js [hunk 2]
```diff
@@ -97,11 +98,17 @@ export default async function onSearchInput({ value, resultsEl, searchInputEl, a
...
-      advancedLink.href = href.toString();
-      resultsEl.replaceChildren(advancedSearchEl);
+      const noResults = await replaceKey('no-results', getConfig());
+      const emptyMessage = createTag('p', {}, noResults);
+      let emptyList = createTag('li', null, emptyMessage);
+      if (advancedSearchEl) {
+        const advancedLink = advancedSearchEl.querySelector('a');
+        const href = new URL(advancedLink.href);
+        href.searchParams.set('q', value);
+        advancedLink.href = href.toString();
```
Comment (overmyheadandbody) [inline libs/blocks/gnav/gnav-contextual-search.js:108]: Would using the `href` property not achieve the same result?

**Review by overmyheadandbody — COMMENTED** [review]
I know that `gnav.js` is sort of in maintenance mode, but the ESLint errors seem quite straightforward to address, I'd vote to fix those so the PR checks pass

File: libs/blocks/gnav/gnav-contextual-search.js (hunk 2 above)
Comment (saugatmalla) [reply libs/blocks/gnav/gnav-contextual-search.js:108]: @overmyheadandbody are you suggesting something like this?
```
const advancedLink = advancedSearchEl.querySelector('a');
const { href } = advancedLink
advancedLink.href = `${href}?q=${value}`
```
In this case, we would be ignoring the existing params.

**Pull request merged**